python3_tools/get_repos.py

```diff
@@ -158,6 +159,32 @@ def is_oep7_compliant(repo):
     # Check to see if openedx.yaml has an oeps section
     return is_oep_compliant(repo, "oep-7")
 
+@cache_to_disk(1)
+def is_oep18_compliant(repo):
+    """
+    OEP-18 is the one about Python Dependency Management
+        -mainly includes implementing make upgrade and placing all the dependencies in requirement directory
+    returns (bool,str): whether it's compliant and reason it's not if bool is false.
+    """
+    return is_oep_compliant(repo, "oep-18")
+
+@cache_to_disk(1)
+def might_be_oep18_compliant(repo):
+    """
+    Use alternate indicators to see if a repo is oep18 complaint. Checks to see if upgrade target is in Makefile
+    
+    returns (bool,str): whether it might compliant and reason we think so if it's true.
+    """
+    try:
+        makefile_contentfile= repo.get_contents("Makefile")
+        makefile_content_file_string = makefile_contentfile.decoded_content.decode('utf-8')
+        search_output = re.search("upgrade:", makefile_content_file_string)
+        if search_output:
+            return [True, "upgrade target exists in Makefile"]
```

Comment:
Why are you using lists instead of tuples?

Reply:
Yep, tuples would be better